src/Evolution/Systems/NewtonianEuler/Limiters/Minmod.cpp

```diff
@@ -0,0 +1,283 @@
+    const tnsr::I<DataVector, VolumeDim, Frame::Logical>& logical_coords,
+    const std::array<double, VolumeDim>& element_size,
+    const Scalar<DataVector>& det_inv_logical_to_inertial_jacobian,
+    const EquationsOfState::EquationOfState<false, ThermodynamicDim>&
+        equation_of_state,
+    const std::unordered_map<
+        std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>, PackagedData,
+        boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>&
+        neighbor_data) const noexcept {
+  if (UNLIKELY(disable_for_debugging_)) {
+    // Do not modify input tensors
+    return false;
+  }
+
+  // Checks for the post-timestep, pre-limiter NewtonianEuler state, e.g.:
+  const double mean_density = mean_value(get(*mass_density_cons), mesh);
+  ASSERT(mean_density > 0.0,
+         "Positivity was violated on a cell-average level.");
+  if constexpr (ThermodynamicDim == 2) {
+    const double mean_energy = mean_value(get(*energy_density), mesh);
+    ASSERT(mean_energy > 0.0,
+           "Positivity was violated on a cell-average level.");
+  }
+  // End pre-limiter checks
+
+  bool limiter_activated = false;
+
+  if (vars_to_limit_ == NewtonianEuler::Limiters::VariablesToLimit::Conserved) {
+    const ConservativeVarsMinmod minmod(minmod_type_, tvb_constant_,
+                                        disable_for_debugging_);
+    limiter_activated =
+        minmod(mass_density_cons, momentum_density, energy_density, mesh,
+               element, logical_coords, element_size, neighbor_data);
+  } else if (vars_to_limit_ ==
+             NewtonianEuler::Limiters::VariablesToLimit::Characteristic) {
+    limiter_activated = characteristic_minmod_impl(
+        mass_density_cons, momentum_density, energy_density, minmod_type_,
+        tvb_constant_, mesh, element, logical_coords, element_size,
+        equation_of_state, neighbor_data);
+  } else {
+    ERROR(
+        "No implementation of NewtonianEuler::Limiters::Minmod for variables: "
+        << vars_to_limit_);
+  }
+
+  if (apply_flattener_) {
+    const Scalar<DataVector> det_logical_to_inertial_jacobian{
+        1.0 / get(det_inv_logical_to_inertial_jacobian)};
+    const auto flattener_action = flatten_solution(
+        mass_density_cons, momentum_density, energy_density, mesh,
+        det_logical_to_inertial_jacobian, equation_of_state);
+    if (flattener_action != FlattenerAction::NoOp) {
+      limiter_activated = true;
+    }
+  }
+
+  // Checks for the post-limiter NewtonianEuler state, e.g.:
+  ASSERT(min(get(*mass_density_cons)) > 0.0, "Bad density after limiting.");
+  if constexpr (ThermodynamicDim == 2) {
+    const auto specific_internal_energy = Scalar<DataVector>{
+        get(*energy_density) / get(*mass_density_cons) -
+        0.5 * get(dot_product(*momentum_density, *momentum_density)) /
+            square(get(*mass_density_cons))};
+    const auto pressure = equation_of_state.pressure_from_density_and_energy(
```

Comment:
can you do the pressure computation inside the `ASSERT`? Maybe wrap this entire if block in an `ifdef SPECTRE_DEBUG`?

Reply:
Good call, I'll do something like this